features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/factories/event/TimelineItemContentFactory.kt

```diff
@@ -57,8 +60,10 @@ class TimelineItemContentFactory @Inject constructor(
             is RoomMembershipContent -> roomMembershipFactory.create(eventTimelineItem)
             is StateContent -> stateFactory.create(eventTimelineItem)
             is StickerContent -> stickerFactory.create(itemContent)
-            is PollContent -> pollFactory.create(itemContent)
-            is PollEndContent -> pollEndFactory.create(itemContent)
+            is PollContent ->
+                if (featureFlagService.isFeatureEnabled(FeatureFlags.Polls)) pollFactory.create(itemContent) else TimelineItemUnknownContent
```

Comment:
I'd have put the check in the PollFactory

Reply:
I was not sure but yeah, let's keep this factory lighter